base/indices.jl

```diff
@@ -377,9 +377,22 @@ Represent an AbstractUnitRange `range` as an offset vector such that `range[i] =
 
 `IdentityUnitRange`s are frequently used as axes for offset arrays.
 """
-struct IdentityUnitRange{T<:AbstractUnitRange} <: AbstractUnitRange{Int}
-    indices::T
+struct IdentityUnitRange{I<:AbstractUnitRange, T} <: AbstractUnitRange{T}
```

Comment:
```suggestion
struct IdentityUnitRange{T, I<:AbstractUnitRange} <: AbstractUnitRange{T}
```

For consistency; people would expect the first parameter to be `eltype`. See also `OffsetArrays. IdOffsetRange`

Reply:
Unfortunately this is breaking as some packages appear to rely on the present signature with the first parameter as the range

Reply:
https://github.com/JuliaApproximation/QuasiArrays.jl/blob/2b06122ac40f95344c01c0b5b328b7765ea3f4dc/src/quasiarray.jl#L106

During the GitHub search, I also found an example that already assumes this PR is merged 😆 (with first parameter to be eltype)

cc: @dlfivefifty

Reply:
Ah, guess I never actually hit that path

Reply:
Looking at https://juliahub.com/ui/RepoSearch?q=IdentityUnitRange%7B&r=true, it appears that there are a couple of other packages aside from OffsetArrays and FFTViews. I wonder if this is to be treated as an internal detail that packages should not rely upon?

Reply:
Should it still at least have `T<:Integer`?

Reply:
And add a Julia 2.0 comment about changing the order of the type parameters.

Reply:
Another option would be to make
```julia
julia> struct MyRange{I<:AbstractUnitRange{<:Integer}} <: AbstractUnitRange{eltype{I}}
           inds::I
       end
ERROR: TypeError: in Type{...} expression, expected UnionAll, got a value of type typeof(eltype)
Stacktrace:
 [1] top-level scope
   @ REPL[19]:1
```
work.